test/com/facebook/buck/apple/AppleBinaryIntegrationTest.java

```diff
@@ -855,6 +855,22 @@ public void testAppleBundleDebugFormatRespectsDefaultConfigSettingNoDebug() thro
         Matchers.equalTo(false));
   }
 
+  @Test
+  public void testBinaryWithFrameworks() throws Exception {
+    assumeTrue(Platform.detect() == Platform.MACOS);
+    ProjectWorkspace workspace = TestDataHelper.createProjectWorkspaceForScenario(
+        this, "apple_binary_with_library_dependencies_as_frameworks", tmp);
+    workspace.setUp();
+
+    BuildTarget target = BuildTargetFactory.newInstance("//TestApp:TestAppBundle#include-frameworks");
+    ProjectWorkspace.ProcessResult result = workspace.runBuckCommand(
+        "build", target.getFullyQualifiedName());
+    result.assertSuccess();
+
```

Comment:
I need to write some more asserts for this. It currently builds but doesn't run because of the aforementioned rpath problem. After I fix that I'll update the test case. The test case is elaborate on mixing frameworks/non frameworks and should give a good impression of the requirements for what I'm trying to achieve.